tests/idris2/basic/basic049/expected

```diff
@@ -15,30 +15,6 @@ Warning: We are about to implicitly bind the following lowercase names.
 You may be unintentionally shadowing the associated global definitions:
   a is shadowing Main.R2.R2.a, Main.R1.R1.a
 
-Fld:76:1--81:38
- 76 | interface Show a => (num : Num a) => MyIface a where -- Some interface with
- 77 |   constructor MkIface
- 78 |                                              -- constraints
- 79 |   data MyData : a -> Type                    -- and a data declaration.
- 80 |   someFunc : a -> a                          -- Constraints are now elaborated as auto implicits (as one would expect)
- 81 |   giveBack : {x : a} -> MyData x -> a        -- (previously as explicit arguments of the interface
-
-Warning: We are about to implicitly bind the following lowercase names.
-You may be unintentionally shadowing the associated global definitions:
-  a is shadowing Main.R2.R2.a, Main.R1.R1.a
-
-Fld:76:1--81:38
- 76 | interface Show a => (num : Num a) => MyIface a where -- Some interface with
- 77 |   constructor MkIface
- 78 |                                              -- constraints
- 79 |   data MyData : a -> Type                    -- and a data declaration.
- 80 |   someFunc : a -> a                          -- Constraints are now elaborated as auto implicits (as one would expect)
- 81 |   giveBack : {x : a} -> MyData x -> a        -- (previously as explicit arguments of the interface
-
-Warning: We are about to implicitly bind the following lowercase names.
-You may be unintentionally shadowing the associated global definitions:
-  a is shadowing Main.R2.R2.a, Main.R1.R1.a
-
```

Comment:
It looks like these warnings were occurring in hints for constraints. I'm not sure if they are useful, but it seems odd

Reply:
> I'm not sure if they are useful, but it seems odd

To me these warning look completely okay, since both interface definition and interface implementations seem to shadow the top-level `a`s

Reply:
Yes, but in the following code `MyIface0`, `MyIface1` and `MyIface2` produce 0, 1 and 2 warnings respectively. This means that one warning per constraint is generated. This is because they do not arise from the data the interface is elabored into, but from the constraint hint. This is strange
```idris
a : Type

interface MyIface0 a where
  f0 : a -> Type

interface Show a => MyIface1 a where
  f1 : a -> Type

interface Show a => (num : Num a) => MyIface2 a where
  f2 : a -> Type
```

Reply:
A warning about shadowing is produced when `a` is an unbound implicit, but in the above examples `a` is explicitly bound
```idris
a : Type

-- Produce warning
interface Iface1 (x : a) where

-- No warning
interface Iface2 a (x : a) where
```